### PR TITLE
Add ACE support for Rancher provisioned RKE2/K3S clusters

### PR DESCRIPTION
**Pull request by thedadams**
In order to supoort ACE for RKE2/K3S clusters, the v3 Cluster object
needs to track which v3 Nodes are associated to it. NodesFromCAPI is
used for this purpose. Nodes are added/removed from rancher-operator.

Also, RKE uses the label `controlplane` for the control plane role,
while RKE2/K3S uses `control-plane`. A matchExpresion is added so that
the kube-api-auth daemon set schedules a pod on a node with the
`control-plane` role.

Issue:
https://github.com/rancher/rancher/issues/31604

### CODE REVIEW

File: pkg/api/norman/customization/cluster/actions_kubeconfig.go [hunk 1]
```diff
@@ -64,6 +64,9 @@ func (a ActionHandler) GenerateKubeconfigActionHandler(actionName string, action
 		}
 
 		cfg, err = kubeconfig.ForClusterTokenBased(&cluster, apiContext.ID, apiContext.Request.Host, tokenKey)
+		if err != nil {
+			return err
+		}
```
Comment (rmweir) [inline pkg/api/norman/customization/cluster/actions_kubeconfig.go:69]: An error check is already being done on L73.

**Review by rmweir — COMMENTED** [review]
One comment, otherwise LGTM.

File: pkg/api/norman/customization/cluster/actions_kubeconfig.go (hunk 1 above)
Comment (thedadams) [reply pkg/api/norman/customization/cluster/actions_kubeconfig.go:69]: The `err` variable is being shadowed on line 46, so the value of `err` on line 73 is different.

Comment (dramich) [reply pkg/api/norman/customization/cluster/actions_kubeconfig.go:69]: Trying to hold that error all the way through is a bad pattern, it takes too long to parse this function to determine where those errors are being set and as Donnie found, we can accidentally shadow it and never catch the error. We should be checking them when we create them.

File: pkg/kubeconfig/kubeconfig.go [hunk 2]
```diff
@@ -150,6 +146,17 @@ func ForClusterTokenBased(cluster *managementv3.Cluster, clusterID, host, token
 			}
 			nodes = append(nodes, clusterNode)
 		}
+	} else {
+		for _, v3Node := range cluster.NodesFromCAPI {
+			if v3Node.ControlPlane {
+				nodes = append(nodes, node{
+					ClusterName: v3Node.RequestedHostname,
```
Comment (dramich) [inline pkg/kubeconfig/kubeconfig.go:153]: Is the name format being changed on purpose? I would think it should be same as above to keep the kubeconfig consistent.

Comment (thedadams) [reply pkg/kubeconfig/kubeconfig.go:153]: Yes, the `RequestedHostname` already has the cluster name in it. For example, `donnie-test-cluster-nodepool-pool1-d82r8`.

**Review by dramich — DISMISSED** [review]

**Pull request merged**